src/hooks/api/useApiCalls.ts

```diff
@@ -45,7 +56,76 @@ const useApiCalls = (): ApiCallsContextType => {
     [getValuesByChain]
   );
 
-  return { createType, stateCall };
+  const updateSenderAccountsInformation = useCallback(
+    async (dispatchAccount) => {
+      const formatBalanceAddress = (data: any, api: ApiPromise): BalanceState => {
+        return {
+          chainTokens: data.registry.chainTokens[0],
+          formattedBalance: formatBalance(data.free, {
+            decimals: api.registry.chainDecimals[0],
+            withUnit: api.registry.chainTokens[0],
+            withSi: true
+          }),
+          free: data.free
+        };
+      };
+
+      if (!keyringPairsReady || !keyringPairs.length) {
+        return {};
+      }
+
+      const getAccountInformation = async (sourceRole: any, targetRole: any) => {
+        const {
+          apiConnection: { api: sourceApi },
+          chain: sourceChain,
+          configs: sourceConfigs
+        } = sourceRole;
+        const {
+          apiConnection: { api: targetApi },
+          configs: targetConfigs
+        } = targetRole;
+
+        const accounts = await Promise.all(
```

Comment:
Why you need an `await Promise.all` here?

Reply:
because the iterator is returning an array of promises and what i want to is to fill the array `accounts` with the results. I have removed the `Promise.all` and i got the following typing issue:

```
 Type 'Promise<{ account: { address: string; balance: BalanceState; name: string; }; companionAccount: { address: string; balance: BalanceState; name: string; }; }>[]' is not assignable to type 'DisplayAccount[]'.
```